core/src/main/scala/org/bitcoins/core/wallet/utxo/UTXOSpendingInfo.scala

```diff
@@ -404,32 +551,62 @@ case class P2PKWithTimeoutSpendingInfo(
     }
 }
 
-case class MultiSignatureSpendingInfo(
+sealed trait MultiSignatureSpendingInfo
+    extends RawScriptUTXOSpendingInfoSingle {
+  override def conditionalPath: ConditionalPath =
+    ConditionalPath.NoConditionsLeft
+  override def scriptPubKey: MultiSignatureScriptPubKey
+}
+
+case class MultiSignatureSpendingInfoSingle(
     outPoint: TransactionOutPoint,
     amount: CurrencyUnit,
     scriptPubKey: MultiSignatureScriptPubKey,
-    signers: Vector[Sign],
+    signer: Sign,
     hashType: HashType
-) extends RawScriptUTXOSpendingInfo {
-  require(signers.length >= scriptPubKey.requiredSigs, "Not enough signers!")
+) extends MultiSignatureSpendingInfo
 
-  override def conditionalPath: ConditionalPath =
-    ConditionalPath.NoConditionsLeft
-}
-
-/** Info required for signing a [[ConditionalScriptPubKey]] */
-case class ConditionalSpendingInfo(
+case class MultiSignatureSpendingInfoFull(
     outPoint: TransactionOutPoint,
     amount: CurrencyUnit,
-    scriptPubKey: ConditionalScriptPubKey,
+    scriptPubKey: MultiSignatureScriptPubKey,
     signers: Vector[Sign],
-    hashType: HashType,
-    conditionalPath: ConditionalPath)
-    extends RawScriptUTXOSpendingInfo {
+    hashType: HashType
+) extends RawScriptUTXOSpendingInfo
+    with MultiSignatureSpendingInfo {
+  require(signers.length >= scriptPubKey.requiredSigs,
+          s"Not enough signers!: $this")
+
+  override val requiredSigs: Int = scriptPubKey.requiredSigs
+
+  override def toSingle(signerIndex: Int): MultiSignatureSpendingInfoSingle = {
+    MultiSignatureSpendingInfoSingle(outPoint,
+                                     amount,
+                                     scriptPubKey,
+                                     signers(signerIndex),
+                                     hashType)
+  }
+
+  override def toSingles: Vector[MultiSignatureSpendingInfoSingle] = {
+    signers.take(requiredSigs).map { signer =>
+      MultiSignatureSpendingInfoSingle(outPoint,
```

Comment:
If we had a 2-of-3 multisig, this would only allow us to sign with the first 2 keys

Reply:
ditto above